homeworks/kirkovvalentiniván/7_extract_transform/cripto_investment.py

```diff
@@ -0,0 +1,112 @@
+import requests
+import pandas as pd
+
+class CoingeckoAPI: 
+    
+    def __init__(self, base_url, params):
+        self.base_url = base_url
+        self.params = params 
+    
+        self.raw_data = None
+        self.df = None
+
+    def get_data(self):
+        response = requests.get(self.base_url, params = self.params)
+        if response.status_code == 200:
+            self.raw_data = response.json()
+            print("Data is extracted successfully.")
+        else:
+            raise Exception(f" API error: {response.status_code}, {response.text}")
+
+class ResponseTransformation:    
+
+    def __init__(self, raw_data):
+        
+        if not raw_data:
+            raise ValueError("There is no raw data or something went wrong.")
+        self.df = pd.DataFrame(raw_data)
+        print("Dataframe is created successfully.")
+        
+    def count_missing_values(self):
+        """Count missing values in the DataFrame."""
+        if self.df is None:
```

Comment:
ezek a check-ek innentől kezdve nem kellenek, mert nincs is már `create_dataframe()` method.
az init-ben már raise-eled az errort, ha a raw_data None, szóval a self.df fixen létezni fog valamilyen formában (vagy már az initben hibát dob ha nem tud dataframe-et létrehozni)